MediaBrowser.Common/Extensions/ProcessExtensions.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MediaBrowser.Common.Extensions
+{
+    /// <summary>
+    /// Extension methods for <see cref="Process"/>.
+    /// </summary>
+    public static class ProcessExtensions
+    {
+        /// <summary>
+        /// Gets a value indicating whether the associated process has been terminated using
+        /// <see cref="Process.HasExited"/>. This is safe to call even if there is no operating system process
+        /// associated with the <see cref="Process"/>.
+        /// </summary>
+        /// <param name="process">The process to check the exit status for.</param>
+        /// <returns>
+        /// True if the operating system process referenced by the <see cref="Process"/> component has
+        /// terminated, or if there is no associated operating system process; otherwise, false.
+        /// </returns>
+        public static bool HasExitedSafe(this Process process)
+        {
+            try
+            {
+                return process.HasExited;
+            }
+            catch (InvalidOperationException)
+            {
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously wait for the process to exit.
+        /// </summary>
+        /// <param name="process">The process to wait for.</param>
+        /// <param name="timeout">The duration to wait before cancelling waiting for the task.</param>
+        /// <returns>True if the task exited normally, false if the timeout elapsed before the process exited.</returns>
+        public static async Task<bool> WaitForExitAsync(this Process process, TimeSpan timeout)
+        {
+            using (var cancelTokenSource = new CancellationTokenSource(timeout))
```

Comment:
Note that I've removed the validation here to ensure that timeout is > 0. I think it's better to just throw an exception in that case. The existing behavior was to set a timeout of zero which would immediately cancel the task.

I've checked and the existing calls to this method and are all guaranteed to pass > 0.